src/providers/cryptoauthlib/asym_sign.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::Provider;
+use crate::authenticators::ApplicationName;
+use crate::key_info_managers::KeyTriple;
+use log::{error, info};
+use parsec_interface::operations::psa_key_attributes::{EccFamily, Type};
+use parsec_interface::operations::{psa_sign_hash, psa_verify_hash};
+use parsec_interface::requests::{ProviderID, ResponseStatus, Result};
+use rust_cryptoauthlib::AtcaStatus;
+
+impl Provider {
+    pub(super) fn psa_sign_hash_internal(
+        &self,
+        app_name: ApplicationName,
+        op: psa_sign_hash::Operation,
+    ) -> Result<psa_sign_hash::Result> {
+        let key_triple = KeyTriple::new(app_name, ProviderID::CryptoAuthLib, op.key_name.clone());
+        let key_attributes = self.key_info_store.get_key_attributes(&key_triple)?;
+        op.validate(key_attributes)?;
+
+        let mut signature = vec![0u8; rust_cryptoauthlib::ATCA_SIG_SIZE];
+        let key_id = self.key_info_store.get_key_id::<u8>(&key_triple)?;
+        info!("psa_sign_hash_internal - using slot {}", key_id);
+        let result = self.device.sign_hash(
+            rust_cryptoauthlib::SignMode::External(op.hash.to_vec()),
+            key_id,
+            &mut signature,
+        );
+        match result {
+            AtcaStatus::AtcaSuccess => Ok(psa_sign_hash::Result {
+                signature: signature.into(),
+            }),
+            _ => {
+                error!("Sign failed, hardware reported: {}", result);
+                Err(ResponseStatus::PsaErrorHardwareFailure)
+            }
+        }
+    }
+
+    pub(super) fn psa_verify_hash_internal(
+        &self,
+        app_name: ApplicationName,
+        op: psa_verify_hash::Operation,
+    ) -> Result<psa_verify_hash::Result> {
+        let key_triple = self
+            .key_info_store
+            .get_key_triple(app_name, op.key_name.clone());
+        let key_attributes = self.key_info_store.get_key_attributes(&key_triple)?;
+        op.validate(key_attributes)?;
+
+        let key_id = self.key_info_store.get_key_id::<u8>(&key_triple)?;
+        info!("psa_sign_hash_internal - using slot {}", key_id);
```

Comment:
Should maybe be `psa_verify_hash_internal` here

Reply:
Yes, you are right. I am correcting this.